examples/text/plot_document_classification_20newsgroups.py

```diff
@@ -19,99 +19,14 @@
 #         Lars Buitinck
 # License: BSD 3 clause
 
-import logging
-import numpy as np
-from optparse import OptionParser
-import sys
-from time import time
-import matplotlib.pyplot as plt
 
-from sklearn.datasets import fetch_20newsgroups
-from sklearn.feature_extraction.text import TfidfVectorizer
-from sklearn.feature_extraction.text import HashingVectorizer
-from sklearn.feature_selection import SelectFromModel
-from sklearn.feature_selection import SelectKBest, chi2
-from sklearn.linear_model import RidgeClassifier
-from sklearn.pipeline import Pipeline
-from sklearn.svm import LinearSVC
-from sklearn.linear_model import SGDClassifier
-from sklearn.linear_model import Perceptron
-from sklearn.linear_model import PassiveAggressiveClassifier
-from sklearn.naive_bayes import BernoulliNB, ComplementNB, MultinomialNB
-from sklearn.neighbors import KNeighborsClassifier
-from sklearn.neighbors import NearestCentroid
-from sklearn.ensemble import RandomForestClassifier
-from sklearn.utils.extmath import density
-from sklearn import metrics
-
-
-# Display progress logs on stdout
-logging.basicConfig(level=logging.INFO, format="%(asctime)s %(levelname)s %(message)s")
-
-op = OptionParser()
-op.add_option(
-    "--report",
-    action="store_true",
-    dest="print_report",
-    help="Print a detailed classification report.",
-)
-op.add_option(
-    "--chi2_select",
-    action="store",
-    type="int",
-    dest="select_chi2",
-    help="Select some number of features using a chi-squared test",
-)
-op.add_option(
-    "--confusion_matrix",
-    action="store_true",
-    dest="print_cm",
-    help="Print the confusion matrix.",
-)
-op.add_option(
-    "--top10",
-    action="store_true",
-    dest="print_top10",
-    help="Print ten most discriminative terms per class for every classifier.",
-)
-op.add_option(
-    "--all_categories",
-    action="store_true",
-    dest="all_categories",
-    help="Whether to use all categories or not.",
-)
-op.add_option("--use_hashing", action="store_true", help="Use a hashing vectorizer.")
-op.add_option(
-    "--n_features",
-    action="store",
-    type=int,
-    default=2**16,
-    help="n_features when using the hashing vectorizer.",
-)
-op.add_option(
-    "--filtered",
-    action="store_true",
-    help=(
-        "Remove newsgroup information that is easily overfit: "
-        "headers, signatures, and quoting."
-    ),
-)
-
-
-def is_interactive():
-    return not hasattr(sys.modules["__main__"], "__file__")
-
-
-# work-around for Jupyter notebook and IPython console
-argv = [] if is_interactive() else sys.argv[1:]
-(opts, args) = op.parse_args(argv)
-if len(args) > 0:
-    op.error("this script takes no arguments.")
-    sys.exit(1)
+# %%
+# Some choices on the analysis
+# -----------------------------
 
-print(__doc__)
-op.print_help()
-print()
+N_FEATURES = 2**16
+USE_HASHING = False
+SELECT_CHI2 = False
```

Comment:
I think this is an `int` in the original example, used to configure `SelectKBest`:

```suggestion
# If greater than 1, we use `SelectKBest` for feature selection
SELECT_CHI2_K = 0
```

Also I think `SELECT_CHI2_K` is a slightly better name to make it explicit that it is used to set `k` in `SelectKBest`.

Reply:
Yes, but False stills work as a default value. I find the suggested option a tiny but confusing (we are not quite selecting "0" features).

I'm happy to apply it, but then I need to change the name of the parameter further down in the script.

Reply:
I think we need to somehow say that `SELECT_CHI2` can be an `int`. Maybe a comment is good enough?

```suggestion
# If an int, then `SelectKBest` sets `k` to `SELECT_CHI2`
SELECT_CHI2 = False
```

Reply:
I added a comment (sorry, I hadn't seen that you had done a suggestion, so I wrote my own comment)